File: ios2/frames/mapper/AuthMapperImpl.swift
//
//  AuthMapperImpl.swift
//  kofero
//
//  Created by Mitchell Drew on 1/2/24.
//

import Foundation
import presenter
import SwiftyJSON


class AuthMapperImpl: AuthMapper {
    func getAuthToken(jsonString: String) -> String {
        return JSON(parseJSON: jsonString)["token"].stringValue
    }
}

File: and/app/src/main/java/ro/kofe/map/AuthMapperImpl.kt
package ro.kofe.map

import com.google.gson.Gson
import ro.kofe.model.response.RegisterAuthResponse
import ro.kofe.presenter.provider.AuthMapper

class AuthMapperImpl(private val gson: Gson) : AuthMapper {
    override fun getAuthToken(jsonString: String) =
        gson.fromJson(jsonString, RegisterAuthResponse::class.java).token
}